### PR TITLE
Overhaul nav

### PR DESCRIPTION
**Pull request by sipec**
- add bottom section for sign out, help
- remove "Get M$", twitter
  - Get Mana added to profile - which is where people will naturally click on to add money since it has your balance
     image

- remove from mobile nav: search, profile, charity, refer friend
  - You can still get to refer a friend from add funds modal, or from share link on a market
  - Charity moved to labs
  - Search and profile are already in the bottom nav
- remove metamore so groups and labs won't be jealous

Screen Shot 2022-11-05 at 12 25 43 AM
Screen Shot 2022-11-05 at 12 26 42 AM
Screen Shot 2022-11-05 at 12 26 50 AM

### CODE REVIEW

File: web/components/portfolio/portfolio-value-section.tsx [hunk 1]
```diff
@@ -267,3 +275,22 @@ export function TimeSelectionButton(props: {
...
+  if (!userId || user?.id !== userId) return null
+
+  return (
+    <>
+      <Button
+        className="ml-auto self-start"
+        color="gray-outline"
+        onClick={() => setOpen(true)}
+      >
+        Add {ENV_CONFIG.moneyMoniker}
```
Comment (jahooma) [inline web/components/portfolio/portfolio-value-section.tsx:291]: "Add" is a little unclear IMO. "Get" or "Get more" is better

File: common/util/array.ts [hunk 2]
```diff
@@ -1,23 +1,13 @@
...
-    if (Array.isArray(el)) {
-      array.push(...el)
-    } else if (el) {
-      array.push(el)
-    }
-  }
-
-  return array
+export function buildArray<T>(...params: FalseyValueArray<T>[]) {
+  return compact(flattenDeep(params)) as T[]
```
Comment (jahooma) [inline common/util/array.ts:10]: Nice. Didn't know about `compact`

File: common/util/array.ts [hunk 3]
```diff
@@ -1,23 +1,13 @@
...
 
 export function filterDefined<T>(array: (T | null | undefined)[]) {
   return array.filter((item) => item !== null && item !== undefined) as T[]
 }
 
-export function buildArray<T>(
-  ...params: (T | T[] | false | undefined | null)[]
-) {
-  const array: T[] = []
+type FalseyValueArray<T> = T | false | undefined | null | FalseyValueArray<T>[]
```
Comment (jahooma) [inline common/util/array.ts:7]: To be complete, I'd add `| 0 | ''`

File: web/components/nav/sidebar.tsx [hunk 4]
```diff
@@ -108,122 +95,77 @@ const logout = async () => {
...
-    { name: 'Groups', href: '/groups' },
-    { name: 'Refer a friend', href: '/referrals' },
-    { name: 'Charity', href: '/charity' },
-    { name: 'Labs', href: '/labs' },
-    { name: 'Discord', href: 'https://discord.gg/eHQBNBqXuh' },
-    signOut
+    loggedIn && { name: 'Groups', href: '/groups', icon: UsersIcon },
+    // { name: 'Charity', href: '/charity', icon: HeartIcon },
+    // loggedIn && { name: 'Get M$', href: '/add-funds', icon: CashIcon },
+    { name: 'Labs', href: '/labs', icon: BeakerIcon }
```
Comment (jahooma) [inline web/components/nav/sidebar.tsx:153]: I think Refer a friend is important to keep. I've heard from users that referred other users by finding this exact option in the sub menu.

If anything we should put this option in more places. Not take it away, even just for mobile.

File: web/components/nav/sidebar.tsx [hunk 5]
```diff
@@ -108,122 +95,77 @@ const logout = async () => {
...
-    { name: 'Discord', href: 'https://discord.gg/eHQBNBqXuh' },
-    { name: 'Help & About', href: 'https://help.manifold.markets/' },
-    {
-      name: 'Sign out',
-      href: '#',
-      onClick: logout,
-    }
+    { name: 'Blog', href: 'https://news.manifold.markets' },
+    { name: 'Discord', href: 'https://discord.gg/eHQBNBqXuh' }
+    // { name: 'Twitter', href: 'https://twitter.com/ManifoldMarkets' },
```
Comment (jahooma) [inline web/components/nav/sidebar.tsx:138]: I think you could just delete the code for the commented out options.

**Review by jahooma — APPROVED** [review]
This is great, Sinclair! I like the code quality improvements, and elimination of the "more" menu for mobile.

Left a couple comments.

**Comment by ingawei** [comment]
Help & About repeats when signed out?
Screen Shot 2022-11-06 at 9 32 02 PM

**Comment by ingawei** [comment]
I like that sign out is accessible but I was thinking of doing some sidebar things in Cycle 2 that might overwrite these changes :( 
I'm planning on:
- making a Mana page that combines add mana, refer a friend and charities pages
- more button just adds more to the sidebar instead of having it in a dropdowny situation
Feel free to push for now but just know I might be making changes down the road!

**Review by ingawei — APPROVED** [review]

File: common/util/array.ts (hunk 2 above)
Comment (sipec) [reply common/util/array.ts:10]: fyi - can also do this in vanilla js like `params.flat(Infinity).filter(Boolean)` but typescript hates this

**Comment by sipec** [comment]
> I was thinking of doing some sidebar things in Cycle 2 that might overwrite these changes :(

No worries! I just did this in a fit of hyperfocus because it bothered me too much haha

**Pull request merged**